sdk/azcore/internal/exported/request.go

```diff
@@ -102,24 +102,28 @@ func (req *Request) OperationValue(value interface{}) bool {
 
 // SetBody sets the specified ReadSeekCloser as the HTTP request body.
 func (req *Request) SetBody(body io.ReadSeekCloser, contentType string) error {
-	// Set the body and content length.
-	size, err := body.Seek(0, io.SeekEnd) // Seek to the end to get the stream's size
-	if err != nil {
-		return err
+	var err error
+	var size int64
+	if body != nil {
+		size, err = body.Seek(0, io.SeekEnd) // Seek to the end to get the stream's size
+		if err != nil {
+			return err
+		}
 	}
 	if size == 0 {
-		body.Close()
-		return nil
-	}
-	_, err = body.Seek(0, io.SeekStart)
-	if err != nil {
-		return err
-	}
-	req.Raw().GetBody = func() (io.ReadCloser, error) {
-		_, err := body.Seek(0, io.SeekStart) // Seek back to the beginning of the stream
-		return body, err
+		// treat an empty stream the same as a nil one: assign req a nil body
+		body = nil
+	} else {
+		_, err = body.Seek(0, io.SeekStart)
+		if err != nil {
+			return err
+		}
+		req.Raw().GetBody = func() (io.ReadCloser, error) {
+			_, err := body.Seek(0, io.SeekStart) // Seek back to the beginning of the stream
+			return body, err
+		}
 	}
-	// keep a copy of the original body.  this is to handle cases
+	// keep a copy of the body argument.  this is to handle cases
```

Comment:
For the case where we're clearing the body, should we remove the content-type and content-length headers?  Or at the very least the content-type header seeing there's no content.  This might make the case for a separate `RemoveBody()` (or similarly named) API.

Reply:
With this, we'll clear content-type by setting its value to "" and set content-length to "0". The length is redundant, but I believe still correct. We could of course remove it instead when the given stream has no content.

I went with this approach over adding a `RemoveBody()` because that method would be redundant alongside `SetBody()`. It seems intuitive, to me anyway, to remove a request's body by setting it empty. This approach also fixes the quirky current behavior of `SetBody()`, which silently ignores an empty stream input when the request already has a body. We should do something about that in any case. If we don't overwrite the current body, we should return an error.

Reply:
Good points, works for me (we should doc this though).